### PR TITLE
Fixes poly's ghost SM spam

### LINKED ISSUE #34814: Poly's ghost can be consumed infinitely by the SM crystal
**Reproduction:** Dump poly into the SM room, it will walk into the crystal and die, or you can throw it into it. Then its ghost will stay in the chamber and constantly waltz into the SM crystal and trigger admin messages, and spawn tons and tons of ashes in the chamber. You can pretty much run the SM without emitters this way though.
Also, poly's ghost changes sex in the fluff text messages from time to time XD
2018-01-23 00_52_44-space station 13

To be clear, I didn't do that on the TG server, just on localhost while testing another bug.

### PR DESCRIPTION
**Pull request by ShizCalev**
Also resolve the issue with any other mobs that are in GODMODE.

Fixes #34814

### CODE REVIEW

**Comment by MrStonedOne** [comment]
This kills tactical chairs

**Comment by ExcessiveUseOfCobblestone** [comment]
I am ShizCalev, destroyer of FUN

**Comment by 81Denton** [comment]
Goodbye Poly engine, was nice while it lasted.

**Comment by Buggy123** [comment]
Breaks fun :-1: DNM

**Comment by DaedalusGame** [comment]
don't worry, we can still make vending machine engines

**Comment by JohnMcLaime** [comment]
True power of ghost poly comes from the fact that it can be implanted with tracking device, that lets you to lock on it, as if it was a beacon, pair that with poly's headset slot and you have great validhunting potential.

**Comment by AlexTheSysop42** [comment]
tactical chairs?

**Comment by Buggy123** [comment]
Something living hitting the SM makes a metric fuckton of radiation, maybe he means sitting in a chair staring at the SM, getting a nice tan?

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -521,7 +521,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
 	dust_mob(user, cause = "hand")
 
 /obj/machinery/power/supermatter_shard/proc/dust_mob(mob/living/nom, vis_msg, mob_msg, cause)
-	if(nom.incorporeal_move)
+	if(nom.incorporeal_move || nom.status_flags & GODMODE)
```
Comment (Cyberboss) [inline code/modules/power/supermatter/supermatter.dm]: This is a good fix and all, but poly's ghost DOES have incorporeal_move set, so the fact that it gets past here bamboozles me

Comment (ShizCalev) [reply code/modules/power/supermatter/supermatter.dm]: Attack by doesn't call dust_mob, but consume instead

Comment (Cyberboss) [reply code/modules/power/supermatter/supermatter.dm]: why tf is poly attacking the shard? dumb bird

Comment (ExcessiveUseOfCobblestone) [reply code/modules/power/supermatter/supermatter.dm]: Poly has engineering's best interests at heart, even in passing.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**